core-tests/src/test/java/org/mule/MuleMessageDataTypePropagationTestCase.java

```diff
@@ -344,43 +343,49 @@ public void setsDefaultInboundScopePropertyDataType() throws Exception
     }
 
     @Test
-    public void setsDefaultInvocationPropertyDataType() throws Exception
+    public void setsDefaultFlowVariableDataType() throws Exception
     {
-        DefaultMuleMessage muleMessage = new DefaultMuleMessage(TEST, muleContext);
-        muleMessage.setInvocationProperty(TEST_PROPERTY, TEST);
+        MuleEvent muleEvent = MuleTestUtils.getTestEvent(TEST, muleContext);
+        muleEvent.setFlowVariable(TEST_PROPERTY, TEST);
 
-        assertDefaultPropertyDataType(muleMessage, PropertyScope.INVOCATION);
+        DataType<?> actualDataType = muleEvent.getFlowVariableDataType(TEST_PROPERTY);
```

Comment:
Extract this lines into a assertVariableDataType, similar to assertPropertyDataType